src/CurlUtil.cc

```diff
@@ -1039,6 +1050,81 @@ CurlWorker::Run() {
     m_op_map.clear();
 }
 
+bool
+CurlWorker::RefreshCacheToken() {
+    auto [success, contents] = RefreshCacheTokenStatic(m_token_file, m_logger);
+    if (success && !contents.empty()) {
+        m_cache_token = contents;
+    }
+    return success;
+}
+
+std::pair<bool, std::string>
+CurlWorker::RefreshCacheTokenStatic(const std::string &token_location, XrdCl::Log *log) {
+    if (token_location.empty()) {
+        return {true, ""};
+    }
+
+    std::string line;
+    std::ifstream fhandle;
+    fhandle.open(token_location);
+    if (!fhandle) {
+        log->Error(kLogXrdClPelican, "Cache token location is set (%s) but failed to open (worker PID %d): %s", token_location.c_str(), getthreadid(), strerror(errno));
+        return {false, ""};
+    }
+
+    std::string result;
+    while (std::getline(fhandle, line)) {
+        rtrim(line);
+        ltrim(line);
+        if (line.empty() || line[0] == '#') {continue;}
```

Comment:
Are these token files a common file type where we'd expect `#` to mark comments, or are we defining that here?

Reply:
It's not exactly an RFC but this is what we follow:

https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md

Reply:
It's not exactly an RFC but this is what we follow:

https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md